### PR TITLE
Goad4windows

### PR DESCRIPTION
**Pull request by svetterIO**
Porting of goad.sh and check.sh to PowerShell variants incl. following new features:
- Task: purge (destroy VM without confirmation Y/N for each machine)
- added vagrant tasks: reload, snapshot, resume, suspend, validate
- optional Kali VM via -k $True/$False/O (only operate on Kali)
- Enable GUI on Provider via -g $True/$False or 1/0
- Typo fixed for GOAD-Light VM names.

Info: Due to porting for windows EOL conversion for bash scripts was required and it is recommended to do this at the repository level rather than in everybody's global Git configuration.

Warning:
Successful provisioning tested only for VMware on Windows 10. VirtualBox: hanging or unreachable VMs. Proxmox: not tested. Providers are kept for testing purposed only.

### CODE REVIEW

**Comment by Mayfly277** [comment]
thanks a lot, very nice work, but i will need some time to test this ^^

**Comment by NeffIsBack** [comment]
Really cool! My plan was to set up the Lab on my new Win machine the next days. As I always used virtualbox so far I would like to use that there as well. Any idea why the machines are hanging? I think experienced that as well in the past, but couldn't resolve it. Did you already dig into it? Maybe I can get it to work.

**Comment by NeffIsBack** [comment]
So in order to get a fresh version of the Lab onto my Host i switched to vmware.

My feedback:
First i tried with podman, apparently there are several bugs and i didn't manage to get that stuff working. podman-cli 5.0.0 is somehow incompatible to the desktop version for now and downgrading also didn't work (after removing some files manually that didn't get cleaned up i still ran into errors).
After switching to docker i had some problems with my network. I couldn't connect to the lab, although the vagrant configuration looked all fine. Apparently the issue was, that my windows network adapter was still configured to use dhcp, so after assigning an ip with subnet (and solving some port conflicts, seems like a vagrant issue) it worked.
Also a nice addition would be to add all the options from the sh script.
Having the kali option is also really nice! Thanks for that. Although it looks like it configures the vmnet2 adapter 2x times?
image


TLDR;

- I would recommend using docker instead of podman as it seems more stable to me
- The issue with the own windows adapter needs to be tested and perhaps fixed with some powershell workaround

**Comment by NeffIsBack** [comment]
Weird, tried building it on my laptop. There the vmnet2 adapter had an ip address of 192.168.156.1. After changing that to 192.168.56.2 it worked flawlessly as well

**Comment by NeffIsBack** [comment]
Another bug while provisioning the SCCM lab: 
The `ANISBLE_COMMAND` here: https://github.com/svetterIO/GOAD/blob/fc5e7e6bc3d7173bcfda10ecca9bc03771894c7c/goad.ps1#L112
was somehow broken, threw an error and didn't play one of the ansible playbooks. I solved it by replacing `"ANSIBLE_COMMAND='ansible-playbook -i ../ad/$LAB/data/inventory -i ../ad/$LAB/providers/$PROVIDER/inventory'` with `LAB=$lab PROVIDER=$provider` which would therefore be picked up by the provisioning script.

**Comment by Mayfly277** [comment]
will not be include as v3 will have kind of windows install solutions.
Please fell free to do a pull request on the new version i i miss something
